core/src/main/java/software/amazon/awssdk/core/client/BaseClientHandler.java

```diff
@@ -155,7 +155,8 @@ protected ExecutionContext createExecutionContext(SdkRequest originalRequest) {
                                                                      .request(originalRequest)
                                                                      .build())
                                .executionAttributes(executionAttributes)
-                               .signerProvider(overrideConfiguration.advancedOption(SdkAdvancedClientOption.SIGNER_PROVIDER))
+                               .signer(overrideConfiguration.advancedOption(SdkAdvancedClientOption.SIGNER))
+                               .signerContext(overrideConfiguration.advancedOption(SdkAdvancedClientOption.SIGNER_CONTEXT))
```

Comment:
should signerContext and signer also be an option in requestOverrideConfig (if it's not already, haven't gotten that far)

Reply:
At the moment, they are not exposed in requestOverrideConfig. I don't think its a common use case to override signer on a single request. As this might be rare, should we wait for now and add the support if we get a request for it? Do you think its common enough to add it now?